symengine/tests/basic/test_basic.cpp

```diff
@@ -383,6 +383,13 @@ TEST_CASE("compare: Basic", "[basic]")
     r2 = mul(x, z);
     CHECK(r1->compare(*r2) == -1);
     CHECK(r2->compare(*r1) == 1);
+//    CHECK(r1->compare(*r2) == -1);
+//    CHECK(r2->compare(*r1) == 1);
+
+    r1 = mul(y, x);
+    r2 = mul(x, z);
+//    CHECK(r1->compare(*r2) == -1);
+//    CHECK(r2->compare(*r1) == 1);
```

Comment:
Hash is platform dependent. That's why in `printer.cpp` we introduce a new map with a different less operator.
```C++
//! Less operator `(<)` using cmp:
struct RCPBasicKeyLessCmp {
    //! true if `x < y`, false otherwise
    bool operator() (const RCP<const Basic> &x, const RCP<const Basic> &y) const {
        if (x->__eq__(*y)) return false;
        return x->__cmp__(*y) == -1;
    }
};
```
which now I think is not enough for data like `x + y + z*x*y + x*y`